### PR TITLE
Move to OR-Tools assets

### PR DESCRIPTION
**Pull request by braktar**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,37 +6,32 @@ env:
 
 install: skip
 
+before_script:
+  - chmod +x ci-utils/ortools/asset_url.sh
```
Comment (alain-andre) [inline .travis.yml:10]: You can commit this file chmoded, to avoid doing it every time in the script.

File: README.md [hunk 2]
```diff
@@ -49,4 +37,6 @@ Dev
 
 After dev the code shall be formatted according to the style file of the project with the following command:
 
-clang-format -i *.cc *.h
+    clang-format -i *.cc *.h
+
+To build OR-Tools image, modify *.travis.yml* `ORTOOLS_VERSION` variable and push on branch ortools.
```
Comment (alain-andre) [inline README.md:42]: I think you can remove this line

File: ci-utils/ortools/asset_url.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+echo $ORTOOLS_VERSION
+if [ $ORTOOLS_VERSION = 'v7.1' ]; then
+  export ORTOOLS_URL="https://github.com/google/or-tools/releases/download/v7.1/or-tools_debian-9_v7.1.6720.tar.gz";
+elif [ $ORTOOLS_VERSION = 'v7.5' ]; then
+  export ORTOOLS_URL="https://github.com/google/or-tools/releases/download/v7.5/or-tools_debian-10_v7.5.7466.tar.gz";
+elif [ $ORTOOLS_VERSION = 'v7.6' ]; then
+  export ORTOOLS_URL="https://github.com/google/or-tools/releases/download/v7.6/or-tools_debian-10_v7.6.7691.tar.gz";
+else
+  echo "Unknown OR-Tools version"
+fi
```
Comment (senhalil) [inline ci-utils/ortools/asset_url.sh:11]: This is a perfect case to use case :) 
https://bash.cyberciti.biz/guide/The_case_statement

File: Makefile [hunk 4]
```diff
@@ -1,39 +1,36 @@
...
 TUTORIAL=resources
 
-
-include $(OR_TOOLS_TOP)/Makefile
-CFLAGS := $(CFLAGS) -DOR_TOOLS_VERSION=\"$(OR_TOOLS_VERSION)\"
+CXXFLAGS := -I $(OR_TOOLS_TOP)/include
 
 # For debugging uncomment the next line. -isystem prevents warnings rooted in or-tools library appearing in our compilation
-# CFLAGS := $(CFLAGS) -ggdb -Og -DDEBUG -fsanitize=address -Wall -Wextra -Wshadow -Wunreachable-code -Winit-self -Wmissing-include-dirs -Wswitch-enum -Wfloat-equal -Wundef -isystem$(OR_TOOLS_TOP)/. -isystem$(OR_TOOLS_SOURCES)/gen -isystem$(OR_TOOLS_TOP)/dependencies/install/include -isystem$(OR_TOOLS_TOP)/dependencies/install/include/coin -DUSE_CBC -DUSE_CLP -DUSE_GLOP -DUSE_BOP
+# CFLAGS := $(CFLAGS) -ggdb -Og -DDEBUG -fsanitize=address -Wall -Wextra -Wshadow -Wunreachable-code -Winit-self -Wmissing-include-dirs -Wswitch-enum -Wfloat-equal -Wundef -isystem$(OR_TOOLS_TOP)/. -DUSE_CBC -DUSE_CLP -DUSE_GLOP -DUSE_BOP
```
Comment (senhalil) [inline Makefile:8]: Forgot to rename CFLAGS -> CXXFLAGS

After renaming can you try to compile activating this line to see if it works as intended (i.e., it doesn't show warnings originated form or-tools library)?

File: Dockerfile [hunk 5]
```diff
@@ -1,23 +1,23 @@
...
+ARG ORTOOLS_URL=${ORTOOLS_URL}
 ARG REGISTRY=${REGISTRY:-registry.mapotempo.com/}
 
-# Install ORTools
-FROM ${REGISTRY}mapotempo/ortools:${ORTOOLS_VERSION} as ortools
-
+# Final image
 # Build wrapper
+FROM debian:latest
+ARG ORTOOLS_URL
```
Comment (senhalil) [inline Dockerfile:7]: This line looks like extra :thinking: is it needed?

File: .travis.yml [hunk 6]
```diff
@@ -6,37 +6,32 @@ env:
...
-        - echo "$REGISTRY_PASSWORD" | docker login -u "$REGISTRY_USERNAME" --password-stdin ${REGISTRY}
-        - export IMAGE_NAME=${REGISTRY}mapotempo/ortools:${ORTOOLS_VERSION}
-        - cd docker/ortools
-        - docker build --build-arg ORTOOLS_VERSION=${ORTOOLS_VERSION} -f ./Dockerfile -t ${IMAGE_NAME} .
-        - docker push ${IMAGE_NAME}
-
     - # Build tag image
       if: tag ~= /^v.*/
       script:
+        - source docker/ortools/asset_url.sh
```
Comment (senhalil) [inline .travis.yml:30]: I am not sure but I think `docker` should be `ci-utils` right?

File: Dockerfile (hunk 5 above)
Comment (braktar) [reply Dockerfile:7]: The variables defined before a FROM are default, but unknown if not called again in the current context

File: .travis.yml [hunk 7]
```diff
@@ -9,34 +9,39 @@ install: skip
...
   include:
     - # Build Dev specific image
-      if: branch = dev
+      if: type NOT IN (pull_request) AND branch = dev
       script:
-        - echo "> Building image for dev perpose with or-tools ${ORTOOLS_VERSION}"
+        - echo "> Building image for dev purpose with or-tools ${ORTOOLS_VERSION}"
         - echo "$REGISTRY_PASSWORD" | docker login -u "$REGISTRY_USERNAME" --password-stdin ${REGISTRY}
         - export OWNER=$(echo ${TRAVIS_REPO_SLUG} | sed "s/\/.*//")
         - export IMAGE_NAME=${REGISTRY}mapotempo-ce/optimizer-ortools:${OWNER:-latest}
```
Comment (alain-andre) [inline .travis.yml:17]: We can remove the latest from `${OWNER:-latest}` as *OWNER* is always set.

**Comment by braktar** [comment]
Within Optimizer-API travis, OPTIMIZER_ORTOOLS_VERSION will be "mapotempo", "braktar", "fonsecadeline" or "senhalil" for latest image of each repo (branch dev)
Or it will point a specific version like "v1.3.0".

Every pull request will verify that the image creation is functional but won't create the image.

**Comment by senhalil** [comment]
@fab-girard can you merge this request. It is needed for other PR to use Travis

**Pull request merged**